common/common-client/src/main/java/org/ow2/proactive/authentication/FileLoginModule.java

```diff
@@ -207,22 +219,37 @@ protected boolean logUser(String username, String password, boolean printErrorMe
      * @throws LoginException if login file is not found or unreadable.
      */
     private boolean authenticateUserFromFile(String username, String password) throws LoginException {
+
+
         Properties props = new Properties();
+        PrivateKey privateKey = null;
         try {
+            privateKey = getPrivateKey();
             FileInputStream stream = new FileInputStream(new File(loginFile));
             props.load(stream);
             stream.close();
+        } catch (KeyException e) {
+            throw new LoginException(e.toString());
         } catch (FileNotFoundException e) {
             throw new LoginException(e.toString());
         } catch (IOException e) {
             throw new LoginException(e.toString());
         }
 
         // verify the username and password
-        if (!props.containsKey(username) || !props.get(username).equals(password)) {
+        if (!props.containsKey(username)) {
             return false;
-        } else
+        } else {
+            String encryptedPassword = (String) props.get(username);
+            try {
+                if (!HybridEncryptionUtil.decryptBase64String(encryptedPassword, privateKey, ENCRYPTED_DATA_SEP).equals(password)) {
+                    return false;
+                }
+            } catch (KeyException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1166)